dojo/api_v2/permissions.py

```diff
@@ -184,11 +184,12 @@ def has_permission(self, request, view):
             # engagement_id doesn't exist
             raise serializers.ValidationError("Engagement '%s' doesn''t exist" % engagement_id)
 
-        if check_import_engagement_permission(request.user, product, product_name, engagement_name, auto_create_context):
-            return True
-
-        if check_import_product_permission(request.user, product, product_name, product_type, product_type_name, auto_create_context):
-            return True
+        if not auto_create_context:
+            raise serializers.ValidationError("Engagement '%s' doesn't exist in Product %s" % (engagement_name, product_name))
```

Comment:
This error message is misleading when the product doesn't exist, because it doesn't mention a missing product.